makefile

```diff
@@ -0,0 +1,17 @@
+B=MESON
+D=apps/aligner/test_data
+PREFIX=${CURDIR}/PREFIX
+
+build:
+	ninja -v -C $B
```

Comment:
I'd remove this `makefile`. In a Github repo, the presence of a Makefile will purely confuse people, because literally noone expects a Makefile as a frontend to a meson + ninja command. Lots of people will trip over it, because the standard expectation is that "if it includes a Makefile, it surely there to build the sources"

Reply:
Makes sense @SoapZA ! Although I have to admit, I usually put Makefile in most of my projects, just as a frontend to basic commands that I often run in the project (so that makes the two of us :D!) -> then I don't have to look them up in the README, I just do quick `cat Makefile` and I use them. At the end, if it runs ninja + Meson, it still is building the sources, so I guess people's expectation would still be correct? But I get when you mean with it not building the sources -> it certainly is not doing what you would expect from Makefile.
We could alternatively go with simple bash script that exposes couple of commands, but I find Makefile to be simpler solution.

Reply:
Don't let Soap discourage you from using makefiles for organizing bash. We have heard his strong opinions at work too. It's orthogonal to how to write `meson.build`.

The `makefile` was actually for elucidation. It shows how to run things when you're not yet familiar with meson. A readme would be fine.

But I'm with you, Martin: A makefile adds a lot of convenience for a developer, and when it's simple, it's like a little tutorial. A `bash` file can be the easiest thing to read, but a `make` file lets you run sections of bash easily.

Make has many valuable use-cases. I don't even like running tests in Meson, personally. Meson hides too much. It's not as opaque as Cmake, but it's still annoying. Integration tests are way easier to write, to understand, and to debug in a `makefile`, and the parallelism is then more obvious. I like unit-tests in Meson (e.g. your `runTests`), and of course you need Meson to build any test-related source-code.

Anyway, this is all academic. The hard part will be to use meson/ninja in travisci. Look [here](https://github.com/open-source-parsers/jsoncpp/blob/master/.travis.yml) and [here](https://github.com/open-source-parsers/jsoncpp/tree/master/.travis_scripts) (contributed by users). Maybe Soap has a better suggestion. (On Mac, `brew install meson ninja` works great.) That's a lot of work for something pretty simple. It's no worse than what you have for **cmake**, but it's still a lot.

I think you should start with *some* `meson.build`, then become familiar with it, then get it into TravisCI, then simplify.

Reply:
Ha, thank you for the interesting discussion! I believe the difference in view here is coming from level of experience in certain fields -> more you know about something, more it seems obvious / redundant to do certain things, and different expectations you have.
I have learned to greatly appreciate @SoapZA 's advice since he seems to have quite more C/C++ experience (especially regarding the build process) than me, and is willing to explain it to me in simple terms when needed :). This is also very important to me because his views and perspective are certainly in sync with how many users of Edlib also think and see things (those more experience with C++). On the other hand, one part of them is certainly more like me :D, so at the end I try to combine the best from both worlds :D.

Using meson/ninja in travisci -> good point, that is the next step certainly. Thanks for links, after we merge this PR I will create an issue to handle that and then try to get to it when I catch some time to focus on it properly.
Thanks again for all the effort!

Reply:
Re Travis:
https://pypi.org/project/meson/
https://pypi.org/project/ninja/ (yes, ninja is packaged on pypi. This obviously abuses PyPI, but means you can avoid installing ninja via `brew`)

Reply:
Getting `ninja` from **pypi** is a nice idea. I will experiment with that.

Reply:
Well, **pypi** definitely worked for `ninja`, so that simplifies our linux setup. But for OSX it turns out that we need **homebrew** for `clang-format` anyway.